tagstudio/src/qt/ts_qt.py

```diff
@@ -1323,57 +1323,60 @@ def get_frame_contents(self, index=0, query: str = ""):
             len(self.frame_dict[query]),
         )
 
-    def filter_items(self, query: str = ""):
-        if self.lib:
-            # logging.info('Filtering...')
-            self.main_window.statusbar.showMessage(
-                f'Searching Library for "{query}"...'
-            )
-            self.main_window.statusbar.repaint()
-            start_time = time.time()
-
-            # self.filtered_items = self.lib.search_library(query)
-            # 73601 Entries at 500 size should be 246
-            all_items = self.lib.search_library(query)
-            frames: list[list[tuple[ItemType, int]]] = []
-            frame_count = math.ceil(len(all_items) / self.max_results)
-            for i in range(0, frame_count):
-                frames.append(
-                    all_items[
-                        min(len(all_items) - 1, (i) * self.max_results) : min(
-                            len(all_items), (i + 1) * self.max_results
-                        )
-                    ]
-                )
-            for i, f in enumerate(frames):
-                logging.info(f"Query:{query}, Frame: {i},  Length: {len(f)}")
-            self.frame_dict[query] = frames
-            # self.frame_dict[query] = [all_items]
-
-            if self.cur_query == query:
-                # self.refresh_frame(self.lib.search_library(query))
-                # NOTE: Trying to refresh instead of navigating forward here
-                # now creates a bug when the page counts differ on refresh.
-                # If refreshing is absolutely desired, see how to update
-                # page counts where they need to be updated.
-                self.nav_forward(*self.get_frame_contents(0, query))
-            else:
-                # self.nav_forward(self.lib.search_library(query))
-                self.nav_forward(*self.get_frame_contents(0, query))
-            self.cur_query = query
-
-            end_time = time.time()
-            if query:
+    
+    def filter_items(self, query: str = "", override: bool = True):
+        if self.lib and (override or (not (self.last_search_string == query))):
+            if(override or (not (self.last_search_string == query))):        
```

Comment:
This second if statement can be removed since its already checked above

Reply:
And self.last_search_string should be replaced with self.cur_query since it already keeps track of the same thing and is used below

Reply:
Sure, done that!